vllm/transformers_utils/image_processor.py

```diff
@@ -1,6 +1,33 @@
 from typing import cast
 
 
+def get_video_processor(
+    processor_name: str,
+    trust_remote_code: bool = False,
+):
+    """
+    Gets a processor for the given model name via HuggingFace.
+    """
+    from transformers import AutoProcessor
+
+    try:
+        processor = AutoProcessor.from_pretrained(processor_name)
+        video_processor = processor.video_processor
+
+    except ValueError as e:
+        if not trust_remote_code:
+            err_msg = (
+                "Failed to load the processor. If the processor is "
+                "a custom processor not yet available in the HuggingFace "
+                "transformers library, consider setting "
+                "`trust_remote_code=True` in LLM or using the "
+                "`--trust-remote-code` flag in the CLI.")
+            raise RuntimeError(err_msg) from e
+        else:
+            raise e
+    return video_processor
+
+
```

Comment:
AFAIK, `LlavaNextVideoImageProcessor` is essentially an `ImageProcessor` itself, so you should be able to load it from `AutoImageProcessor`.
See https://github.com/huggingface/transformers/blob/87134662f73d5e89bb015531ddd1d4662371d317/src/transformers/models/auto/image_processing_auto.py#L101

Reply:
I have tried that but it ends up loading a `LlavaNextImageProcessor`. Let me find out what is the problem.

Reply:
`AutoImageProcessor` cannot correctly load the `LlavaNextVideoImageProcessor`. Here's what happens in https://github.com/huggingface/transformers/blob/87134662f73d5e89bb015531ddd1d4662371d317/src/transformers/models/auto/image_processing_auto.py#L410-L442
1. It loads `preprocessor_config.json` from huggingface: https://huggingface.co/llava-hf/LLaVA-NeXT-Video-7B-hf/blob/main/preprocessor_config.json
2. The preprocessor of `Llava-next-video` includes an image processor(`LlavaNextImageProcessor`) and a video processor(`LlavaNextVideoImageProcessor`):https://github.com/huggingface/transformers/blob/87134662f73d5e89bb015531ddd1d4662371d317/src/transformers/models/llava_next_video/processing_llava_next_video.py#L66-L67. However in `preprocessor_config.json`, it only specifies `"image_processor_type": "LlavaNextImageProcessor"`.
3. `AutoImageProcessor` first identifies the `image_processor_type` in configuration file, and get `LlavaNextImageProcessor`, instead of `LlavaNextVideoImageProcessor`.

Without modifying the source code in transformers lib, the only way to load the `LlavaNextVideoImageProcessor` is loading the `LlavaNextVideoProcessor` first and extracting from it.

Reply:
Got it, thanks for investigating!